mesh/mesh.go

```diff
@@ -55,17 +55,26 @@ type Mesh struct {
 	// if contextual validity changed for blocks in that layer, we need to
 	// double-check whether we have applied the correct block for that layer.
 	minUpdatedLayer atomic.Value
+
+	// used for caching hare certificate when the node sync data from peers.
+	// a synced certificate cannot be verified before the node learns the beacon value
+	// of the epoch. any certificate passed on by the syncer will stay here until
+	// beacon is available.
+	certCache    map[types.LayerID][]*types.Certificate
```

Comment:
i don't think that it is ok to add more mutable state randomly across the code. is there a way to refactor it in a better way?

can syncer download ballots and blocks before downloading opinions on them?

Reply:
ok. i modified syncer to fetch layer data and layer opinions separately.
the layer opinion is part of the state sync, after beacon is available.